### PR TITLE
[IOTDB-5284] Fix some code smells

### PR DESCRIPTION
**Pull request by BUAAserein**
### Fix some code smells in server [iotdb-server], please see in [JIRA5284](https://issues.apache.org/jira/browse/IOTDB-5284)

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java [hunk 1]
```diff
@@ -193,10 +193,10 @@ public class IoTDBConfig {
   private int walBufferQueueCapacity = 50;
 
   /** Size threshold of each wal file. Unit: byte */
-  private volatile long walFileSizeThresholdInByte = 10 * 1024 * 1024;
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java:196]: Why not use `10 * 1024 * 1024L`?

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java [hunk 2]
```diff
@@ -193,10 +193,10 @@ public class IoTDBConfig {
...
+  private volatile long walFileSizeThresholdInByte = 10 * 1024L * 1024;
 
   /** Size threshold of each checkpoint file. Unit: byte */
-  private volatile long checkpointFileSizeThresholdInByte = 3 * 1024 * 1024;
+  private volatile long checkpointFileSizeThresholdInByte = 3 * 1024L * 1024;
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java:199]: same

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java [hunk 3]
```diff
@@ -206,13 +206,13 @@ public class IoTDBConfig {
    * this, wal can flush this memtable to disk, otherwise wal will snapshot this memtable in wal.
    * Unit: byte
    */
-  private volatile long walMemTableSnapshotThreshold = 8 * 1024 * 1024;
+  private volatile long walMemTableSnapshotThreshold = 8 * 1024L * 1024;
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java:209]: same

File: server/src/main/java/org/apache/iotdb/db/consensus/statemachine/DataRegionStateMachine.java [hunk 4]
```diff
@@ -325,7 +326,7 @@ public List<Path> getSnapshotFiles(File latestSnapshotRootDir) {
           region.getDatabaseName(),
           region.getDataRegionId(),
           e);
-      return null;
+      return Collections.emptyList();
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/consensus/statemachine/DataRegionStateMachine.java:329]: You need to check for upper-level dependencies on null, such as changing `== null` to `isEmpty` or something like that

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java (hunk 1 above)
Comment (BUAAserein) [reply server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java:196]: Already changed

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java (hunk 2 above)
Comment (BUAAserein) [reply server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java]: Already changed

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java (hunk 3 above)
Comment (BUAAserein) [reply server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java]: Already changed

File: server/src/main/java/org/apache/iotdb/db/consensus/statemachine/DataRegionStateMachine.java (hunk 4 above)
Comment (BUAAserein) [reply server/src/main/java/org/apache/iotdb/db/consensus/statemachine/DataRegionStateMachine.java]: Already removed

**Review by SpriCoder — APPROVED** [review]
LGTM~

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**